Source/Scene/Batched3DModel3DTileContent.js

```diff
@@ -501,14 +509,18 @@ define([
         this._model.shadows = this._tileset.shadows;
         this._model.debugWireframe = this._tileset.debugWireframe;
 
+        // Set to rebuild model shaders
+        if (this.shadersDirty) {
+            this._model.shouldRegenerateShaders = true;
+        }
+
         // Update clipping planes
         var tilesetClippingPlanes = this._tileset.clippingPlanes;
-        var modelClippingPlanes = this._model.clippingPlanes;
         if (defined(tilesetClippingPlanes)) {
-            tilesetClippingPlanes.clone(modelClippingPlanes);
-            modelClippingPlanes.enabled = tilesetClippingPlanes.enabled && this._tile._isClipped;
-        } else if (defined(modelClippingPlanes) && modelClippingPlanes.enabled) {
-            modelClippingPlanes.enabled = false;
+            // Dereference the clipping planes from the model if they are irrelevant.
+            // Link/Dereference directly to avoid ownership checks.
+            // This will also trigger synchronous shader regeneration to remove or add the clipping plane and color blending code.
+            this._model._clippingPlanes = (tilesetClippingPlanes.enabled && this._tile._isClipped) ? tilesetClippingPlanes : undefined;
```

Comment:
Slight problem, I think a `clippingPlanesDirty` toggle on `model` doesn't allow a way to tell the model that clipping should be disabled because it's irrelevant. That was part of the reasoning behind referencing/dereferencing.

Reply:
Clarified offline, we can have `Model.js` only update its own `_clippingPlaneState` when it owns the clipping plane collection, otherwise it'll rely on b3dm and i3dm to do that, hence the need for the `clippingPlanesDirty` toggle.

Reply:
Another chat offline: consensus is that the method where b3dm and i3dm handle more state stuff for `Model.js` requires more state variables attached to `Model` prototype. Seems like it should be cleaner to have `b3dm` and `i3dm` continue to do their dereference/re-reference thing.